sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -803,16 +814,19 @@ __ESIMD_API simd<Tx, N> atomic_update(Tx *p, simd<unsigned, N> offset,
 /// @tparam Tx The vector element type.
 /// @tparam N The number of memory locations to update.
 /// @param p The USM pointer.
-/// @param offset The vector of 32-bit offsets in bytes.
+/// @param offset The vector of 32-bit or 64-bit offsets in bytes.
 /// @param src0 The additional argument.
 /// @param mask Operation mask, only locations with non-zero in the
 ///   corresponding mask element are updated.
 /// @return A vector of the old values at the memory locations before the
 ///   update.
 ///
-template <atomic_op Op, typename Tx, int N>
-__ESIMD_API simd<Tx, N> atomic_update(Tx *p, simd<unsigned, N> offset,
-                                      simd<Tx, N> src0, simd_mask<N> mask) {
+template <atomic_op Op, typename Tx, int N, typename Toffset>
+__ESIMD_API std::enable_if_t<std::is_same_v<Toffset, uint32_t> ||
```

Comment:
same here an in other places - template parameter checking approach should be consistent in all APIs.